nrespipe/trace_orders.py

```diff
@@ -0,0 +1,1090 @@
+                                          center=guess_row_index, halfwidth=column_halfwidth)
+    centroid = get_centroid(row_indices, intensities)
+    row_indices, intensities = get_column(image, reference_column,
+                                          center=centroid, halfwidth=column_halfwidth)
+    centroid = get_centroid(row_indices, intensities)
+
+    centroids_neighbor = follow_fiber(image, centroid, reference_column, column_halfwidth, baffle_clip=baffle_clip)
+    polynomial_neighbor = fit_polynomial(np.arange(image.shape[1]), centroids_neighbor, degree=degree)
+
+    red_fiber_centroids, blue_fiber_centroids, red_polynomials, blue_polynomials = assign_fiber_colors(
+        polynomial_initial, polynomial_neighbor, centroids_initial, centroids_neighbor, reference_column)
+
+    plot_first_polynomial(image, red_fiber_centroids, red_polynomials, blue_fiber_centroids, blue_polynomials,
+                          debug_plots=debug_plots)
+    plot_first_polynomial_residuals(image, red_fiber_centroids, red_polynomials, blue_fiber_centroids, blue_polynomials,
+                                    debug_plots=debug_plots)
+
+    red_peak = get_peak_flux(image, red_polynomials, reference_column, column_halfwidth)
+    order_separation_initial = get_initial_order_separation(image, red_peak, blue_polynomials, reference_column,
+                                                            column_halfwidth)
+
+    red_fiber_centroids, blue_fiber_centroids, red_polynomials, blue_polynomials = find_other_orders(
+        image, order_separation_initial, red_fiber_centroids, blue_fiber_centroids, red_polynomials,
+        blue_polynomials, reference_column, column_halfwidth, baffle_clip, degree, n_orders, verbose)
+
+    plot_all_polynomials(image, red_fiber_centroids, blue_fiber_centroids, red_polynomials, blue_polynomials,
+                         debug_plots=debug_plots)
+    plot_all_polynomial_residuals(image, red_fiber_centroids, red_polynomials, blue_fiber_centroids, blue_polynomials,
+                                  debug_plots=debug_plots)
+
+    make_ascii(out_file, in_file, header, red_polynomials, blue_polynomials, every)
+
+    if return_objects:
+        return red_fiber_centroids, blue_fiber_centroids, red_polynomials, blue_polynomials
+
+
+def find_first_neighbor_fiber(image, centroids_initial, reference_column, column_halfwidth):
+    """
+    Executes a blind searches for the neighbor fiber (i.e., within the same order) to the initial fiber.
+
+    Parameters
+    ----------
+    image : 2D array
+    centroids_initial : 1D array
+    reference_column : int
+    column_halfwidth : int
+
+    Returns
+    -------
+    neighbor_fiber_row_indices : 1D array
+    neighbor_fiber_intensities : 1D array
+    """
+    fiber_hunt_offset = 2 + column_halfwidth
+    fiber_hunt_column_length = 20
+    centroid_initial = centroids_initial[reference_column]
+
+    candidate_1_indices, candidate_1_intensities = get_column(
```

Comment:
Centroid_initial has been NaN which crashes everything.

Reply:
I can try to reproduce and correct the issue if you send me the files you used.

Reply:
elpnrs02-fl30-20180620-0006-w00.fits.fz which should be available on the archive.lco.global is the file in question. Let me know if you can't find it.

Reply:
I can't find it. Here's the search criteria I used. https://archive.lco.global/?q=a&RLEVEL=&PROPID=&INSTRUME=&OBJECT=&SITEID=elp&TELID=&FILTER=&OBSTYPE=LAMPFLAT&EXPTIME=&BLKUID=&REQNUM=&basename=&start=2018-06-19%2000%3A00&end=2018-06-21%2023%3A00&id=&public=true

Reply:
Should be available now. https://archive.lco.global/?q=a&RLEVEL=&PROPID=&INSTRUME=&OBJECT=&SITEID=&TELID=&FILTER=&OBSTYPE=&EXPTIME=&BLKUID=&REQNUM=&basename=fl30&start=2018-06-01%2000%3A00&end=2018-11-30%2023%3A59&id=&public=true

Reply:
Hmm. I still can't find it. I did find elp1m002-fl30-20180620-0006-w00.fits though.

Reply:
It appears that in this case the SNR in the blue orders is ~3 in image elp1m002-fl30-20180620-0006-w00.fits. Certainly this is too low for decent flat fielding. Or is your concern that currently my function does not exit gracefully when given poor quality data?

Reply:
I'm more worried about failing gracefully.

Reply:
I'm reasonably confident that the routine will work as-is for flats that don't have super low SNR. But, there's always going to be bad data of some kind or another eventually. What would you like my code to do if it seems that the order tracing is failing for whatever reason? Should it return a value indicating its success or failure that is then used to determine whether the pipeline should continue to operate on the image? Or should it print to some log file and then sys.exit()?

Reply:
It should print to the logs and the return back up the stack. not necessarily just exit. I just don't want random crashes on bad data. You can even raise a useful exection like I have done under process_nres_file  in tasks.

Reply:
Ok. I'll put in some exceptions in the subroutines. Can you point me to an example of how to write to the logs?